### DISCUSSION TITLE
How to retrieve a list of VM/Host Groups using govmomi instead of govc command line

### DISCUSSION THREAD

**Discussion by edubois10** [vmware/govmomi#3031] [Community Help]
Hi There,

I need to create a kubernetes controller that will check for VMs in a given VM Group. If I want to do this using govc I just need to
`govc cluster.group.ls -name <nameOfTheGroup>` and I will get the vms in the group, similarly if I want to add vms I would use the  `govc cluster.group.change` command.

However, I struggle to understand how govc and govmomi are linked in that matter. I tried to retrieve a ClusterComputeResource or and from there try to get the vmGroup. But I don't know how to get from the ClusterComputeResource to a vmGroup. I tried with ResourcePools but I have the same issue.

```
func main() {
    example.Run(fun(ctx context.Context, c *vim25.Client) error {

    // Get the group
    group := "examplegroup"

    // Create a new finder
    f := find.NewFinder(client.Client, true)

    // Get the group
    res, err := f.ClusterComputeResourceList(ctx, group)
    if err != nil {
        log.Fatal(err)
    }
    /* From here is only testing and where I am blocked.
    if len(res) == 0 {
        log.Fatalf("Group '%s' not found", group)
    }

    var vms []object.VirtualMachine
    for _, child := range res[0].ChildEntity {
        if child.Reference().Type == "VirtualMachine" {
            vms = append(vms, child.(object.VirtualMachine))
        }
    }

    // Retrieve the VM name
    var vmlist []mo.VirtualMachine
    pc := property.DefaultCollector(client.Client)
    err = pc.Retrieve(ctx, vms, []string{"name"}, &vmlist)
    if err != nil {
        log.Fatal(err)
    }

    for _, vm := range vmlist {
        fmt.Printf("VM: %s\n", vm.Name)
    }
    */
}
```

Thank you in advance for your help.

**Comment by dougm** [comment]
See: https://github.com/vmware/govmomi/issues/3033
